Hangfire.Couchbase/BucketManagerExtensions.cs

```diff
@@ -0,0 +1,36 @@
+﻿using System;
+using System.Linq;
+using Couchbase;
+using Couchbase.Management;
+
+namespace Hangfire.Couchbase
+{
+    public static class BucketManagerExtensions
+    {
+        public static IResult CreateAndCheckN1qlPrimaryIndex(this IBucketManager @this, string indexName)
+        {
+            var result = @this.CreateN1qlPrimaryIndex(indexName, false);
+            AndCheckN1QlIndex(@this, indexName, result);
+            return result;
+        }
+
+        public static IResult CreateAndCheckN1qlIndex(this IBucketManager @this, string indexName, params string[] fields)
+        {
+            var result = @this.CreateN1qlIndex(indexName, false, fields);
+            AndCheckN1QlIndex(@this, indexName, result);
+            return result;
+        }
+
+        private static void AndCheckN1QlIndex(IBucketManager @this, string indexName, IResult result)
```

Comment:
I don't think we will need this method. `IResult` already has `Success` and `Exception` we just use the flag and rethrow the exception.

Reply:
The problem is that lack of permission doesn't actually return an exception. ALSO, if the index already exists, Success will be false then, too.

Reply:
then let's remove the extension class and have the check method moved inside the `CouchbaseStorage` class. This method will accept array of index names and will throw exception if any of them do not exists. 

We will throw our own exception `HangfireCouchbaseMissingIndexException` - will need to create.

Reply:
I was going through the Couchbase github repo and it always returns the exception. Let me check if the exception is not returned due to permission.

Reply:
You can see what was going on (and maybe where I went wrong) at about 38 minutes in: https://www.youtube.com/watch?v=0UhHX_-y1bA including the exception behavior that I'm seeing